### ISSUE TITLE
The inference and training code doesn't work properly

### ISSUE THREAD

**Issue by JinxVision** [intel-isl/DeepLagrangianFluids#3]
1. After I run the pretrained model：
`cd scripts
./run_network.py --weights pretrained_model_weights.h5 \
                 --scene example_scene.json \
                 --output example_out \
                 --write-ply \
                 train_network.py`
I got the right output ( after visualized it). However, I found the code worked on CPU although the GPU memory usage is about 100%. Hence the inference time was about 6 hours.

2. After I run the training code:
`cd datasets
./create_data.sh
cd ../
python scripts/train_network.py`
It works on GPU, however, the loss was NaN. I install the latest  SPlisHSPlasH (v2.7.0), it has exchanged Static/DynamicBoundarySimulator by SPHSimulator which can handle dynamic and static scenes. So I generate the data by SPHSimulator. I have also tried to use older version of SPlisHSPlasH (v2.5.0), however, 'create_data.sh' cannot run successfully with DynamicBoundarySimulator.
I wonder if the generated training data wrong, so could you please provide your generated data or check the code.

**Comment by benjaminum** [comment]
Hi,

1. Can you check which operations have been mapped to which device?
  You can enable to log this information by putting ```tf.debugging.set_log_device_placement(True)``` at the beginning of the script.
  Which GPU are you using? 

2. Please try to generate the data with SPlisHSPlasH v2.4.0. Make sure to compile it in Release mode. The changes to the scene format in v2.5.0 and later are not compatible with our data generation script at the moment. Please also update to the latest commit. We fixed a bug with the creation of the scene dirs. Make sure that the output data dirs do not exist before running ```create_data.sh``` because ```create_physics_scenes.py``` avoids overwriting existing data.

**Comment by JinxVision** [comment]
1. After print the log, i find that the open3d layers are built on CPU. It works fine after building it on GPU.
2. I try to generate the data with SPlisHSPlasH v2.4.0, however, the generating speed is extremely slow. (6 hours for generating a scene.)

**Comment by benjaminum** [comment]
On my system with 32 threads it takes about 1 minute per scene. Maybe there is a problem with SPlisHSPlasH not using all threads.

**Comment by JinxVision** [comment]
So how to use all threads

**Comment by benjaminum** [comment]
Have you checked that SPlisHSPlasH only uses a single thread?
If this is the case you could check if OpenMP is enabled for your compiler. For gcc the flag is ```-fopenmp```. SPlisHSPlasH v2.4.0 sets the flag here https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/c78e738d441dded16335007129eb33f1d08024c1/CMake/Common.cmake#L22

**Comment by alex-wy** [comment]
> Have you checked that SPlisHSPlasH only uses a single thread?
> If this is the case you could check if OpenMP is enabled for your compiler. For gcc the flag is `-fopenmp`. SPlisHSPlasH v2.4.0 sets the flag here https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/c78e738d441dded16335007129eb33f1d08024c1/CMake/Common.cmake#L22

I follow ur advice to install SPlisHSPlasH v2.4.0 and git the lastest commit.
But I still got a problem when I try to generate the data. The data generation program got stuck and did not show any message after generating the "sim_0001". 
I checked the system and found that the  program is running with 32 threads.

**Comment by JinxVision** [comment]
We generate data using SPlisHSPlasH with 8 threads, however, it still takes about 6 hours  per scene.

**Comment by benjaminum** [comment]
I am not sure if I can help with this problem but if you want I can have a look at the ```log/SPH_log.txt``` that is stored in each scene directory.

**Comment by JinxVision** [comment]
[SPH_log.txt](https://github.com/intel-isl/DeepLagrangianFluids/files/4649930/SPH_log.txt)

This is the log of sim0001

**Comment by benjaminum** [comment]
The number of particles, triangles and so on is the same as in my log. The only difference is the time information. For the simulation step I have 
```
Average time: SimStep: 6.82791 ms
```
Are the examples bundled with SPlisHSPlasH also that slow for you? If yes, maybe the SPlisHSPlasH developers can help.

**Comment by benjaminum** [comment]
Closing this thread since the problem is mostly related to SPlisHSPlasH. Don't hesitate to reopen if there is new information or if you have a question.

**Comment by JinxVision** [comment]
It takes so long to simulate data. Can you provide your generated data?
